Library/Homebrew/utils/shared_audits.rb

```diff
@@ -124,6 +166,21 @@ def gitlab(user, repo)
     "GitLab repository too new (<30 days old)"
   end
 
+  def codeberg(user, repo)
+    metadata = codeberg_repo_data(user, repo)
+
+    return if metadata.nil?
+
+    return "Codeberg fork (not canonical repository)" if metadata["fork"]
+    if (metadata["forks_count"] < 30) && (metadata["stars_count"] < 75)
+      return "Codeberg repository not notable enough (<30 forks and <75 stars)"
```

Comment:
I'd like to contest the notability requirements for Codeberg. As a source code forge, it's not used nearly as much as GitHub/GitLab but the projects may still be consumed a lot by users through other channels, e.g. downstream packagers.

Codeberg lacks much of the social "buzz" that GitHub has (e.g. trending list, personal feed, "awesome" lists) and so discoverability of projects (and the social notion of giving projects "stars") within Codeberg is limited. The concept of stars exist in Codeberg but I think generally people just don't star things they use, either because they never need to go look at the source for it, or they're just not inclined to interact with their source code forge as a social media platform.

Also, some projects have also moved to Codeberg from their prior homes elsewhere, so it's not uncommon that the notability stats don't reflect actual usage or popularity.

Reply:
@alebcay Can you suggest some alternative values? It doesn't seem reasonable to say there should be no notability requirements on GitHub (or GitLab) but it is reasonable to say these values should vary per platform.

Reply:
Here's a look at the stars/forks for some formulae we already have which are currently hosted on Codeberg. I've also included the Repology spread to give some idea of how widely it's packaged downstream, as another metric of adoption/usage, and also our formula stats (30/90/365 days) for formula install/install on request.

| Formula | Project | Stars | Forks | Repology spread | Formula install (30/90/365 days) | Formula install on request (30/90/365 days) |
| --- | --- | --- | --- | --- | --- | --- |
| [`ipmitool`](https://formulae.brew.sh/formula/ipmitool) | https://codeberg.org/IPMITool/ipmitool | 3 | 7 | [30](https://repology.org/project/ipmitool/information) | 275/1052/2820 | 275/1049/2807 |
| [`jdupes`](https://formulae.brew.sh/formula/jdupes) | https://codeberg.org/jbruchon/jdupes | 7 | 0 | [20](https://repology.org/project/jdupes/information) | 181/324/634 | 181/324/634 |
| [`libid3tag`](https://formulae.brew.sh/formula/libid3tag) | https://codeberg.org/tenacityteam/libid3tag | 1 | 0 | [43](https://repology.org/project/libid3tag/information) | 843/1470/2689 | 142/228/346 |
| [`libkeccak`](https://formulae.brew.sh/formula/libkeccak) | https://codeberg.org/maandree/libkeccak | 2 | 0 | [6](https://repology.org/project/libkeccak/information) | 182/564/1258 | 4/18/63 |
| [`sha3sum`](https://formulae.brew.sh/formula/sha3sum) | https://codeberg.org/maandree/sha3sum | 4 | 0 | [5](https://repology.org/project/sha3sum/information) | 261/856/1744 | 262/855/1745 |
| [`smake`](https://formulae.brew.sh/formula/smake) | https://codeberg.org/schilytools/schilytools | 20 | 10 | [8](https://repology.org/project/smake/information) | 197/268/599 | 136/168/277 |
| [`sound-touch`](https://formulae.brew.sh/formula/sound-touch) | https://codeberg.org/soundtouch/soundtouch | 15 | 16 | [33](https://repology.org/project/soundtouch/information) | 23/63/146 | 23/63/146 |
| [`star`](https://formulae.brew.sh/formula/star) | https://codeberg.org/schilytools/schilytools | 20 | 10 | [10](https://repology.org/project/star-tar/information) | 88/127/196 | 88/127/196 |

Notes:
- `jdupes` moved from GitHub to Codeberg only in the past month or so
- `smake`/`star` moved from Sourceforge to Codeberg in the past year or so

Not sure I see any correlation between install counts and star/fork numbers, especially considering e.g. `sound-touch` or `smake`/`star` vs `libid3tag`.

Reply:
> Not sure I see any correlation between install counts and star/fork numbers, especially considering e.g. `sound-touch` or `smake`/`star` vs `libid3tag`.

@alebcay Sounds like at least one star would be a sensible requirement and we can drop forks. Thoughts?

Reply:
I think that's sufficiently lax, although I will note that it's possible for the project owner to star their own repo (or for the formula submitter to star the repo). So maybe this is not too different from just having no requirement anyways.

Reply:
I guess this is why we I think we should still have this requirement for then e.g. at least two stars.

It's not a hard requirement that means we reject everything that doesn't meet it. It's a soft requirement which requires us to take a closer look before merging.

Reply:
Is watch count interesting or relevant here?

Reply:
Added watch count for the projects to the table. If I were to put a requirement on that, maybe 3 watchers?

Reply:
Have added a suggestion below for 3 forks, watchers or stars.

Reply:
To be clear: "or" or "and"?

Reply:
Whatever the code says (which is consistent with what we do for GitHub/GitLab)

Reply:
yeah, i agree that we should lower down the codeberg formulae' notability check, I will update it.